src/sACN/E131Client.cpp

```diff
@@ -0,0 +1,267 @@
+#include "E131Client.h"
+#include "cinder/Log.h"
+#include "cinder/app/App.h"
+
+using namespace dmx;
+using namespace std;
+using asio::ip::udp;
+
+E131Client::E131Client(string ipAddress)
+    : ipAddress(ipAddress)
+{
+    // Universally Unique Identifier generated at:  https://www.uuidgenerator.net/
+    const vector<char> cid = { char(0x71), char(0x2d), char(0xfb), char(0x4c), char(0xe6), char(0xa3), char(0x4f), char(0xeb), char(0xaa), char(0x99), char(0xe5), char(0x94), char(0x84), char(0xb4), char(0x5b), char(0x1d) };
+
+    setSourceName("Sosolimited device");
+    setCid(cid);
+    setLengthFlags();
+    setPriority(200); // Default top priority
+    setUniverse(1);
+
+    // Set default data send interval to be equivalent to 30 Hz
+    if (framerate > 0) {
+        dataSendInterval = 1.0f / framerate;
+    }
+
+    // Create a timer for timing cycling and data sending
+    timer = clock();
+
+    // Connect to UDP endpoint
+    connectUDP();
+}
+
+void E131Client::connectUDP()
+{
+    if (!udpSetup) {
+        auto app = cinder::app::AppBase::get();
+
+        try {
+            auto asioAddress = asio::ip::address_v4::from_string(ipAddress.c_str());
+            auto endpoint = udp::endpoint(asioAddress, destPort);
+
+            asio::error_code errCode;
+
+            _socket = make_shared<udp::socket>(app->io_service());
```

Comment:
This will crash on close, since the app's io_service will be deleted twice (once by the app, who owns the underlying service returned by `io_service()`, and once by the shared_ptr created here).

Consider using a separate io_service exclusive to this client. If it is important to use the application's io_service, don't try to manage the memory.